Codeinwp/neve
# AMP infinite scroll

### Summary
- Based on [their documentation](https://amp.dev/documentation/components/amp-next-page/) I added the infinite scroll option for posts too.
- In the documentation, they are mentioning that the `<amp-next-page>` tag must be included before the body tag closes:

_To prevent shifting page content down, this component must be the last child of the document <body>. If needed, any footer content should be embedded inside the <amp-next-page> tag (by adding a container that has the footer attribute) and will be displayed once no more article suggestions are available._

We cannot do this because the whole site is wrapped in a `<div class="wrapper">` so what I did instead was to wrap the footer in `<amp-next-page>` tags by using hooks.

The infinite scroll with amp works like this: every time you reach the end, it makes a request for another page. To avoid getting duplicates you must add `next-page-hide` attribute on the elements you don't want on the next pages. You can't tell "get only this div and put it inside my container".

Because of this mechanism, I choose to load the infinite scroll only when the sidebar is not present on the blog. If you'd have a really long sidebar, you'll have to load render the posts only after the sidebar was rendered, letting a really big space between the first page and the second one.


### Will affect visual aspect of the product
YES

### Screenshots 

### Test instructions
- Before testing, please have a look at how this work here: https://amp.dev/documentation/examples/components/amp-next-page/
- Activate AMP
- Deactivate other plugins so the amp page would be valid
- Enable infinite scroll on the blog and make sure you don't have the sidebar enabled

- It has a strange behavior so let me explain what you should expect
1. The link will change when you reach other page. You're on page 2, the link will be neve.local/page/2. You scroll down to page 3, the link will update accordingly. You scroll back to page 2, the link will update.
2. When the link looks like neve.local/page/x and you refresh the page, the infinite scroll will not work anymore, the pagination will appear instead. This is the desired behavior.
3. When you scroll really fast, something like this will appear https://vertis.d.pr/UXM6bN. And the infinite scroll will not work anymore. If you don't have a site icon, the icon will display as an empty image.

For any questions regarding "should it be like this", please consult their example and see if you can replicate there the problem.

- 
- 


Closes #2917.
